### PR TITLE
Update ESLint config

### PR DESCRIPTION
**Pull request by cangoektas**
### Changes
- Upgrades dependencies to latest version 
- Adds ESLint 4 as a peer dependency (just to be safe)
- Adds Prettier config to remove `airbnb` formatting rules
- Removes custom formatting rules
- Adds Prettier to the project itself
- Improves readability of rules by using `"off"`, `"warn"` and `"error"` instead of `0`, `1`, and `2`

### Todo's
- [ ] Test locally
- [ ] Think about what other changes we might want

### CODE REVIEW

**Comment by cangoektas** [comment]
@alexbeletsky @sfrdmn @5c077yP I will check if the dependency updates break anything and if we can safely move to ESLint 4. In the meantime, I would ask you guys to think about our linting rules and comment here with your suggestions (or directly update the PR). Maybe we can also have a 10 min session about it 💪

Current changes:

- `import/prefer-default-export`: **off**

**Comment by sfrdmn** [comment]
I'm tired of `react/require-default-props` personally. Also, its use apparently prevents some optimizations in `recompose`. Should just use ES6 default values instead IMO

**Comment by sfrdmn** [comment]
Oh, also we need a rule to catch references to `window.find` and other useless globals ;) It exists... and I could have sworn I integrated it somewhere... maybe it was just on user app or just on control center... ?

**Comment by cangoektas** [comment]
`class-methods-use-this` is also something we need to get rid off

**Comment by 5c077yP** [comment]
let's also remove `import/prefer-default-export`

File: package.json [hunk 1]
```diff
@@ -5,28 +5,32 @@
...
   "dependencies": {
-    "babel-eslint": "^7.2.3",
-    "eslint-config-airbnb": "^15.0.1",
-    "eslint-plugin-import": "^2.3.0",
-    "eslint-plugin-jsx-a11y": "^5.0.3",
-    "eslint-plugin-mocha": "^4.9.0",
-    "eslint-plugin-react": "^7.1.0",
-    "estraverse-fb": "^1.3.1"
+    "babel-eslint": "^8.0.0",
+    "eslint": "^4.7.2",
```
Comment (5c077yP) [inline package.json]: i guess this `eslint` from dependencies can be removed

File: package.json [hunk 2]
```diff
@@ -5,28 +5,32 @@
...
   },
   "homepage": "https://github.com/blogfoster/eslint-config-blogfoster#readme",
   "dependencies": {
-    "babel-eslint": "^7.2.3",
-    "eslint-config-airbnb": "^15.0.1",
-    "eslint-plugin-import": "^2.3.0",
-    "eslint-plugin-jsx-a11y": "^5.0.3",
-    "eslint-plugin-mocha": "^4.9.0",
-    "eslint-plugin-react": "^7.1.0",
-    "estraverse-fb": "^1.3.1"
```
Comment (5c077yP) [inline package.json:27]: why did we remove this `estraverse-fb`, is it not needed anymore?

File: package.json [hunk 3]
```diff
@@ -5,28 +5,32 @@
...
+    "eslint-plugin-mocha": "^4.11.0",
+    "eslint-plugin-react": "^7.4.0"
   },
   "devDependencies": {
-    "eslint": "^3.19.0"
+    "eslint": "^4.7.2",
+    "prettier": "^1.7.0"
+  },
+  "peerDependencies": {
+    "eslint": ">= 4"
```
Comment (5c077yP) [inline package.json:33]: did you check if eslint for and the current rules work with our projects? last time i tried it didn't

Comment (sfrdmn) [reply package.json:33]: Er wait, why is eslint a peer dep now actually?

Comment (5c077yP) [reply package.json:33]: @sfrdmn i think, so that you get a npm error while installing just this package from out-side?

Comment (sfrdmn) [reply package.json:33]: Ah right, makes sense. I forgot we were in the config repo for a second :P

File: package.json (hunk 2 above)
Comment (cangoektas) [reply package.json:27]: not needed anymore

File: package.json (hunk 1 above)
Comment (cangoektas) [reply package.json]: was also a peer dependency

File: package.json (hunk 2 above)
Comment (5c077yP) [reply package.json:27]: ok

File: package.json (hunk 3 above)
Comment (cangoektas) [reply package.json:33]: @5c077yP I didn't yet test this but it's on my Todos's 😉

File: package.json (hunk 1 above)
Comment (5c077yP) [reply package.json]: @cangoektas sorry i dont get it really. now we have eslint as `dependency`, `devDependency` and `peerDependency`. aren't the last two enough?

Comment (5c077yP) [reply package.json]: @cangoektas if you check for example the the [airbnb config](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb/package.json), you see they also just have eslint as dev and peer dependency but not as actual dependency

Comment (sfrdmn) [reply package.json]: Yeah, I guess you don't need it as a dep

Comment (sfrdmn) [reply package.json]: Only dev dep, since we lint this repo itself

Comment (cangoektas) [reply package.json]: huh, you're right I ran it again without it and didn't get the peer dependency warning. Yes, `dev` + `peer` is enough

**Comment by cangoektas** [comment]
would also remove `prefer-arrow-callback` since we have a lot of tests now where we can't use them

**Comment by cangoektas** [comment]
we should also restrict usage of browser globals:

```js
// The ESLint browser environment defines all browser globals as valid,
// even though most people don't know some of them exist (e.g. `name` or `status`).
// This is dangerous as it hides accidentally undefined variables.
// We blacklist the globals that we deem potentially confusing.
// To use them, explicitly reference them, e.g. `window.name` or `window.status`.
var restrictedGlobals = [
  'addEventListener',
  'blur',
  'close',
  'closed',
  'confirm',
  'defaultStatus',
  'defaultstatus',
  'event',
  'external',
  'find',
  'focus',
  'frameElement',
  'frames',
  'history',
  'innerHeight',
  'innerWidth',
  'length',
  'location',
  'locationbar',
  'menubar',
  'moveBy',
  'moveTo',
  'name',
  'onblur',
  'onerror',
  'onfocus',
  'onload',
  'onresize',
  'onunload',
  'open',
  'opener',
  'opera',
  'outerHeight',
  'outerWidth',
  'pageXOffset',
  'pageYOffset',
  'parent',
  'print',
  'removeEventListener',
  'resizeBy',
  'resizeTo',
  'screen',
  'screenLeft',
  'screenTop',
  'screenX',
  'screenY',
  'scroll',
  'scrollbars',
  'scrollBy',
  'scrollTo',
  'scrollX',
  'scrollY',
  'self',
  'status',
  'statusbar',
  'stop',
  'toolbar',
  'top',
];
```

from https://github.com/facebookincubator/create-react-app/blob/master/packages/eslint-config-react-app/index.js

*EDIT* Sorry, @sfrdmn you already mentioned that!

**Comment by sfrdmn** [comment]
@cangoektas Ah yeah, that's exactly it. It _should_ be in _some_ version of the AirBnB config... I thought I saw it, anyway. Maybe we should switch to the `create-react-app` eslint config, really.

**Comment by 5c077yP** [comment]
@cangoektas we should also add https://eslint.org/docs/rules/prefer-destructuring

**Comment by cangoektas** [comment]
Rules that need to be turned off:

- `no-console`
- `jsx-a11y/label-has-for`

**Comment by sfrdmn** [comment]
Also this: https://eslint.org/docs/rules/no-alert :P

**Comment by sodoku** [comment]
I would also like to either remove every warning or make them errors. I don't see the point of warnings if they ignored by everyone.

**Comment by cangoektas** [comment]
Also, a couple of rules I used in `blogfoster-scripts`: https://github.com/blogfoster/blogfoster-scripts/blob/master/config/.eslintrc.js

**Comment by 5c077yP** [comment]
close in favor of #12

**Pull request closed without merging**